### PR TITLE
Moving design docs from the wiki

### PR DESCRIPTION
**Pull request by arschles**
Todo after merging this:

- Redirect the [wiki](https://github.com/gomods/athens/wiki/Proxies-(Zeus)) [pages](https://github.com/gomods/athens/wiki/The-Central-Package-Registry-(Olympus)) to the new proxy.md and registry.md pages

Partially addresses #221

### CODE REVIEW

File: PROXY.md [hunk 1]
```diff
@@ -1,61 +1,68 @@
...
+First and foremost, a proxy exposes the same vgo download protocol as the registry. Since it doesn't have the multi-cloud requirements as the registry does, it supports simpler backend data storage mechanisms. We plan to release a proxy with several backends including:
 
-# Proxy Caches
+- In-memory
+- Disk
+- RDBMS
+- Cloud blob storage
 
-The proxy can be configured to store cached modules in several places:
+Users who want to target a proxy configure their `vgo` CLI to point to the proxy, and then execute commands as normal.
```
Comment (marwan-at-work) [inline PROXY.md]: Should we mention that the configuration is through `GOPROXY` env var? Just in case people wanna test it out

File: PROXY.md [hunk 2]
```diff
@@ -1,61 +1,68 @@
...
-* Disk: this is appropriate for single-node deployments. We don't recommend
-  running an athens proxy on a single node, so we don't recomment using disk
-  storage in a large scale deployment. _This backend is not yet complete_
-* RDBMS: this is appropriate for multi-node deployments. We recommend this
-  storage backend for larger scale deployments. _This backend is not yet complete_
+# Cache Misses
 
-The proxy storage drivers are responsible for storing _all_ information
-concerning a module:
+When a user requests a module `MxV1` from a proxy and the proxy doesn't have `MxV1` in its cache, it first determines whether `MxV1` is private or not private.
```
Comment (marwan-at-work) [inline PROXY.md]: how does it determine that a module is indeed private? just for my own curiosity lol

Comment (robjloranger) [reply PROXY.md]: Based on the module filters set up by the org running their proxy. It's up to them to define what paths host private modules.

File: PROXY.md (hunk 1 above)
Comment (arschles) [reply PROXY.md]: I was thinking of creating a separate section / document for how to test the thing (probably in `DEVELOPMENT.md`) and linking to it from here

Comment (robjloranger) [reply PROXY.md]: I think that's a good idea

**Pull request merged**